internal/httpapp/handler.go

```diff
@@ -0,0 +1,237 @@
+package httpapp
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/goccy/go-json"
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/nats-io/nats.go/jetstream"
+	"github.com/rs/zerolog"
+
+	"github.com/hiphops-io/hops/nats"
+	"github.com/hiphops-io/hops/worker"
+)
+
+type (
+	DoInput struct {
+		JSON    interface{}       `json:"json"`
+		Method  string            `json:"method" validate:"required,oneof=GET HEAD POST PUT DELETE OPTIONS PATCH"`
+		Params  map[string]string `json:"params"`
+		Data    []byte            `json:"data"`
+		Retries int               `json:"retries" validate:"gte=0,lte=3"`
+		Headers map[string]string `json:"headers"`
+		URL     string            `json:"url" validate:"required,http_url"`
```

Comment:
should we name this path to be consistent with our other apis? (but might be wrong kind of consistency).